### ISSUE TITLE
Move DX12's Conversion.h.cpp from Atom_RHI_DX12.Private.Static to Atom_RHI_DX12.Reflect

### ISSUE THREAD

**Issue by moraaar** [o3de/o3de#11359]
Move DX12's Conversion.h.cpp from `Atom_RHI_DX12.Private.Static` to `Atom_RHI_DX12.Reflect`.
It suits better there since all the functions are related to classes in Reflect project and other places (like OpenXR, IDK, KTX) could use them.

This was already done for Vulkan (https://github.com/o3de/o3de/pull/11329) and this task is to do the same for DX12.

Unfortunately is not as straight forward as in Vulkan. This is because Vulkan gem has a target that represents glad-vulkan and handles the vulkan headers (`Gem::Atom_RHI_Vulkan.Glad.Static`), which is super convenient because `Atom_RHI_Vulkan.Reflect` can depend on it and that's it, Conversion.h.cpp will have the vulkan includes and the RHI includes, so it all works.

On DX12 Gem there is no target that represent DX12, it's all embedded in `Atom_RHI_DX12.Private.Static`, so when Conversion.h.cpp is moved to `Atom_RHI_DX12.Reflect` it fails to find the dx12.h and it doesn't have the dx12 structures/enums/etc.

Basically, the same way that `Gem::Atom_RHI_Vulkan.Glad.Static` exports publicly <vulkan/vulkan.h>  (which handles per platform vulkan header inclusion), DX12 gems would need a `Gem::Atom_RHI_DX12.DX12.static`
target (for a lack of a better name), which exposes <RHI/DX12_Platform.h> in an include folder, not in the Source where it's now.
